server/controllers/order.js

```diff
@@ -21,21 +21,37 @@ const newOrderValidation = (order) => {
 };
 
 const getAllOrders = (req, res) => {
-  const ordersList = orderModel.getAllOrders();
-  logger.info('Orders were fetched', ordersList);
-  return res.status(200).json({
-    message: 'Orders were fetched',
-    ordersList,
-  });
+  Order.find()
+    .exec()
+    .then((foundOrders) => {
+      if (!foundOrders) {
+        return res.status(404).json({
+          message: 'There are no orders to fetch',
+        });
+      }
+      logger.info('Fetching all orders');
+      return res.status(200).json({
+        message: 'Orders details',
+        foundOrders,
+      });
+    })
+    .catch((err) => {
+      logger.error(err);
+      return res.status(500).json({
+        message: err,
+      });
+    });
 };
```

Comment:
const getAllOrders = async (req, res) => {
  let foundOrders;
  try {
    foundOrders = await Order.find().exec();
  } catch (e) {
    logger.error(err);
    return res.status(500).json({
      message: err,
    });
  }
  
  if (!foundOrders) {
    return res.status(404).json({
      message: 'There are no orders to fetch',
    });
  }
  logger.info('Fetching all orders');
  return res.status(200).json({
    message: 'Orders details',
    foundOrders,
  });
};